presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -327,7 +329,12 @@ public JdbcOutputTableHandle beginCreateTable(ConnectorTableMetadata tableMetada
                     fromProperties(connectionProperties));
         }
         catch (SQLException e) {
-            throw new PrestoException(JDBC_ERROR, e);
+            try {
+                dropTable(new JdbcTableHandle(connectorId, schemaTableName, catalog, schema, temporaryName));
```

Comment:
Thanks @electrum!  I hadn't seen rollback method. 
I moved implementation to JdbcMetadata, hope it is OK